omod/src/main/java/org/bahmni/module/hip/web/service/FHIRResourceMapper.java

```diff
@@ -27,24 +33,51 @@
+    public FHIRResourceMapper(PatientTranslator patientTranslator, PractitionerTranslatorProviderImpl practitionerTranslatorProvider, MedicationRequestTranslator medicationRequestTranslator, MedicationTranslator medicationTranslator, EncounterTranslatorImpl encounterTranslator) {
         this.patientTranslator = patientTranslator;
         this.practitionerTranslatorProvider = practitionerTranslatorProvider;
         this.medicationRequestTranslator = medicationRequestTranslator;
         this.medicationTranslator = medicationTranslator;
         this.encounterTranslator = encounterTranslator;
-        this.observationTranslator = observationTranslator;
     }
 
     public Encounter mapToEncounter(org.openmrs.Encounter emrEncounter) {
         return encounterTranslator.toFhirResource(emrEncounter);
     }
 
-    public Observation mapToObs(Obs obs) {
-        return observationTranslator.toFhirResource(obs);
+    public DocumentReference mapToDocumentReference(Obs obs) {
+        DocumentReference documentReference = new DocumentReference();
+        try {
+            documentReference.setId(obs.getUuid());
+            documentReference.setStatus(Enumerations.DocumentReferenceStatus.CURRENT);
+            List<DocumentReference.DocumentReferenceContentComponent> documentReferenceContentComponentList = new ArrayList<>();
+            DocumentReference.DocumentReferenceContentComponent documentReferenceContentComponent =
+                    new DocumentReference.DocumentReferenceContentComponent();
+            Attachment attachment = new Attachment();
+            attachment.setContentType(getTypeOfTheObsDocument(obs.getValueText()));
+            byte[] fileContent = Files.readAllBytes(new File("/home/bahmni/document_images/" + obs.getValueText()).toPath());
+            attachment.setData(fileContent);
+            documentReferenceContentComponent.setAttachment(attachment);
+            documentReferenceContentComponentList.add(documentReferenceContentComponent);
+            documentReference.setContent(documentReferenceContentComponentList);
+            return documentReference;
+        } catch (IOException exception) {
+            return documentReference;
+        }
+    }
+
+    private String getTypeOfTheObsDocument(String valueText) {
+        if (valueText == null) return "";
+        String extension = valueText.substring(valueText.indexOf('.') + 1);
+        if (extension.compareTo("jpeg") == 0 || extension.compareTo("jpg") == 0) {
+            return "image/jpeg";
+        } else if (extension.compareTo("png") == 0 || extension.compareTo("gif") == 0) {
+            return "image/" + extension;
+        } else {
+            return "application/pdf";
```

Comment:
Add check for pdf extension too. Bahmni does not stop users from uploading any kind of document. For now we are only focusing on image and pdf.